packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -287,12 +296,69 @@ export class ElectronMainApplication {
         return this.didUseNativeWindowFrameOnStart.get(webContents.id) ? 'native' : 'custom';
     }
 
+    /**
+     * Shows the splash screen, if it was configured. Otherwise does nothing.
+     */
+    protected showSplashScreen(): void {
+        // content must be handed over for splash screen to take effect
+        if (this.config.electron.splashScreenOptions?.content) {
+            console.log('Showing splash screen');
+            const splashScreenOptions = this.config.electron.splashScreenOptions;
+            const content = this.config.electron.splashScreenOptions.content;
+            console.debug('SplashScreen options', splashScreenOptions);
+            app.whenReady().then(() => {
+                const splashScreenBounds = this.determineSplashScreenBounds();
+                const splashScreenWindow = new BrowserWindow({
+                    ...splashScreenBounds,
+                    frame: false,
+                    alwaysOnTop: true,
+                    webPreferences: {
+                        preload: content.endsWith('.js') ? path.resolve(this.globals.THEIA_APP_PROJECT_PATH, 'lib', 'frontend', content).toString() : undefined
```

Comment:
Wouldn't `require.resolve()` be a more robust way to load a javascript file?

Reply:
Also, why are we loading `*.js` files in preload and not like the other files?

Reply:
We can't load Javascript files like regular documents as then we will show their content instead of executing them. We would need to specifically call `executeJavascript` and load it beforehand.

Instead I wanted to support the use case of executing within the preload to allow for customizations like transferring information for a progress bar. However as we don't load anything else it's then difficult to get actual content into the window without customizing `showSplashScreen` itself anyway.

We could offer both `content` and `preload` customizations, but that's maybe overkill. `content` should be enough and I'll remove the preload handling. If an adopter needs more control over the window they can still customize `showSplashScreen`.